primo/opt_model/model_options.py

```diff
@@ -273,27 +264,6 @@ def solver(self):
         """Returns the solver object"""
         return self._solver
 
-    def _pairwise_matrix(self, metric: str):
```

Comment:
This part of the code is now vectorized using `DataFrame` methods. See `get_pairwise_metrics` method in `clustering_utils.py` file. Also, from now on, we construct these matrices only if they are needed for efficiency calculations. Otherwise, they are not constructed.

Reply:
Nicely done and much needed :-)